### PR TITLE
[develop] Changes for Derecho, a new platform

### LINKED ISSUE #884: Add Derecho to supported platforms, as Tier-1 system
## Description

Add support for Derecho system at UCAR as Tier-1 platform. Include standard SRW testing suites. 

## Solution
- Prepare modulefiles
- prepare configuration files for standard testing suites
- prepare modules and environments for running the workflow
- prepare machine files 

## Requirements**
- Use updated software stacks currently used by UFS-Weather-Model
-  rocoto build for Derecho
-  workflow python3/conda environments for the SRW tasks 
- no need for additional data staging, as Derecho shares the filesystem with Cheyenne at UCAR 

## Acceptance Criteria (Definition of Done)
All standard tests successfully run and pass on Derecho using manual and automated Jenkins testings

## Dependencies 

Depends on https://github.com/ufs-community/ufs-weather-model/pull/1836

### PR DESCRIPTION
**Pull request by natalie-perlin**
Modulefile and other configuration files to adapt the SRW to Derecho system.

Software stacks used for testing are hdf5/1.14.0, netcdf/4.9.2-based, similar to those used in  https://github.com/ufs-community/ufs-srweather-app/pull/889.

## DESCRIPTION OF CHANGES: 

Adding Derecho system at UCAR/NCAR at Tier-1 machine.

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## TESTS CONDUCTED: 
All fundamental tests pass. 

- [ ] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [x] derecho.intel
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [x] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:
This PR will resolve the issue 884:
https://github.com/ufs-community/ufs-srweather-app/issues/884
This PR depends on https://github.com/ufs-community/ufs-srweather-app/pull/889 - MERGED

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation: add as a new platform. 
- [ ] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

## CONTRIBUTORS (optional): 

@mark-a-potts 

Fundamental tests are successful.
https://github.com/ufs-community/ufs-srweather-app/pull/894#issuecomment-1699916354

[WE2E_summary_20230823001411.txt](https://github.com/ufs-community/ufs-srweather-app/files/12417696/WE2E_summary_20230823001411.txt)
[WE2E_summary_20230823013603.txt](https://github.com/ufs-community/ufs-srweather-app/files/12417700/WE2E_summary_20230823013603.txt)

### CODE REVIEW

File: etc/lmod-setup.sh
Comment (RatkoVasic-NOAA) [inline etc/lmod-setup.sh]: Was cheyenne left out from list on purpose?

**Review by RatkoVasic-NOAA — APPROVED** [review]
Looks good to me.

File: etc/lmod-setup.csh
Comment (MichaelLueken) [inline etc/lmod-setup.csh]: With Cheyenne getting decommissioned at the end of the year, we should remove cheyenne from the list and only include derecho.

**Review by MichaelLueken — COMMENTED** [review]
@natalie-perlin -  Thanks for opening this PR to allow the SRW App to build and run on Derecho!

Since Cheyenne will be decommissioned at the end of the year and given that the NRAL0032 account is out of resources on Cheyenne, should we keep Cheyenne in the various files still, or would it be best to fully transition to Derecho?

If we fully cut support for Cheyenne and fully transition to Derecho, then the modification made in `ush/get_crontab_contents.py` can be changed so that line 61 would read:

`if MACHINE == "DERECHO"`

which should allow the `Python unittests` to pass (currently, the `Python unittests` are failing in `test_get_crontab_contents` because the `crontab_cmd` is being set as `usr/bin/crontab` rather than `crontab`).

Has an EPIC Platform ticket been created to create a new Derecho pipeline so that we can add Derecho to the .cicd/Jenkinsfile to run the automated tests on the new platform?  If not, please let me know and I can open a ticket for this work.

**Comment by natalie-perlin** [comment]
https://github.com/ufs-community/ufs-srweather-app/pull/826

File: etc/lmod-setup.sh
Comment (natalie-perlin) [reply etc/lmod-setup.sh]: Set it back to "cheyenne". Derecho has another procedure of resetting modules, as added further down in lines 50-51.

File: etc/lmod-setup.csh
Comment (natalie-perlin) [reply etc/lmod-setup.csh]: Derecho option to explicitly reset modules to system defaults is listed further down, lines 43-44. I haven't changed anything for cheyenne yet. A separate PR may needs to be done to clean the entire code and remove Cheyenne as a supported platform.

Comment (natalie-perlin) [reply etc/lmod-setup.csh]: Added a "derecho" option to get_crontab_contents.py:
`if MACHINE == "CHEYENNE" or "DERECHO":`

**Comment by natalie-perlin** [comment]
Got help from CISL Derecho team to identify a runtime error for the MET verification tasks. The error was similar to all of the met binaries used, e.g.:

`/glade/work/epicufsrt/contrib/derecho/hpc-stack/intel-classic-2023.0.0/intel-classic-2023.0.0/met/10.1.2/bin/pb2nc: symbol lookup error: /usr/lib64/libssh.so.4: undefined symbol: EVP_KDF_CTX_new_id, version OPENSSL_1_1_1d`

Solution was to install **libssh** package in miniconda3 and conda environments used during SRW workflow runtime. Solution stemmed from the notes on this page:
https://docs.nersc.gov/development/languages/python/using-python-perlmutter/

```
conda activate <environment>
conda install -c conda-forge libssh
```

**Comment by natalie-perlin** [comment]
All the fundamental tests pass successfully. 
WE2E_summary_20230829182239.txt (for fundamental tests) attached.
```
----------------------------------------------------------------------------------------------------
Experiment name              
                                    | Status    | Core hours used
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              17.36
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              22.76
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE              13.33
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              28.37
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR          COMPLETE              35.06
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0              COMPLETE              31.42
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE              42.29
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             190.59
```
[WE2E_derecho_summary_fundamental.txt](https://github.com/ufs-community/ufs-srweather-app/files/12480632/WE2E_derecho_summary_fundamental.txt)

**Comment by natalie-perlin** [comment]
MET verification tests pass all except for **MET_MET_ensemble_verification_only_vx_time_lag**, 
which requires pulling observation data, and it is attempted to pull via HPSS during the run. Observational data exist on disk, as is set up in ./ush/machine/derecho/yaml in variables TEST_CCPA_OBS_DIR, TEST_MRMS_OBS_DIR, TEST_NDAS_OBS_DIR. 
Need to set up the task not to attempt to access HPSS.

Attached is a log file for all met verification experiments: WE2E_derecho_summary_met_verifications.txt

A brief summary:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
MET_verification                                                   COMPLETE              18.34
MET_verification_only_vx                                           COMPLETE               0.11
MET_ensemble_verification                                          COMPLETE              42.52
MET_ensemble_verification_only_vx                                  COMPLETE               0.65
MET_ensemble_verification_only_vx_time_lag                         DEAD                   0.01
----------------------------------------------------------------------------------------------------
```
[WE2E_derecho_summary_met_verifications.txt](https://github.com/ufs-community/ufs-srweather-app/files/12480627/WE2E_derecho_summary_met_verifications.txt)
 
 
 UPDATE: There is an issue with the MET_ensemble_verification_only_vx_time_lag at the moment affecting all platforms, and it has been filed https://github.com/ufs-community/ufs-srweather-app/issues/900

File: etc/lmod-setup.csh
Comment (natalie-perlin) [reply etc/lmod-setup.csh]: @MichaelLueken -
Adding Derecho to ush/get_crontab_contents.py did not solve the issue of falling unittests.

**Comment by natalie-perlin** [comment]
@MichaelLueken - Derecho could be added .cicd/Jenkinsfile to run the automated tests. Please open a ticket for this.

**Comment by MichaelLueken** [comment]
@natalie-perlin - I have opened ticket [PSD-39](https://jira.epic.oarcloud.noaa.gov/servicedesk/customer/portal/5/PSD-39) for the platform team to create a new Jenkins pipeline for Derecho.  From Kris and Keven, it sounds like EPIC is currently unable to support Apps on Derecho.  I will follow-up with you regarding any change to the ticket's status.

**Comment by MichaelLueken** [comment]
@natalie-perlin - Unfortunately, the PSD-39 ticket has been closed saying that no new pipeline will be created for Derecho at this time.  I'll bring this up during tomorrow's SRW CM meeting, since without a pipeline for Derecho, it will be hard to support the SRW App release on the machine.

**Comment by natalie-perlin** [comment]
@MichaelLueken - I have no access to PSD-39 ticket, is there a reason given for not supporting Derecho?..

**Comment by MichaelLueken** [comment]
@natalie-perlin - Unfortunately, no information was given on the PSD-39 ticket.  The status on the ticket was changed to `Completed` and was resolved as `Won't Do`.  From Slack discussions with Kris, this is what he had to say before I opened the ticket yesterday:

`I asked about Derecho a couple of weeks ago and the answer I got from Keven Blackman was, it was not officially supported yet.  There’s still some political challenges to overcome in getting access to it.`

So, it sounds like there are still some political issues that will need to be addressed before a Jenkins pipeline can be established on Derecho.

**Comment by natalie-perlin** [comment]
It sounds like we could go ahead with including Derecho as a supported platform, especially as Cheyenne is not available for testing for EPIC anymore, and is going away soon.

**Comment by natalie-perlin** [comment]
@MichaelLueken what kind of checks are currently failing that prevent this PR from moving ahead?..

**Comment by MichaelLueken** [comment]
@natalie-perlin - The `Python unittests` that are run via GitHub Actions need to pass before we can move forward with this PR.  Until the `test_get_crontab_contents` unit test that is run on by GitHub Actions passes, we can't move forward with this PR.

**Comment by MichaelLueken** [comment]
@natalie-perlin - It also looks like PR #889 will need to be merged into develop first.  The SRW App fails to build on machines outside of Derecho because the `build_*.lua` files aren't pointing to the `netcdf/4.9.2` stack locations, so `netcdf/4.9.2` isn't being found, leading to the failure.

**Comment by RatkoVasic-NOAA** [comment]
@MichaelLueken @natalie-perlin 
I'm reviewing #889 right now.

**Comment by MichaelLueken** [comment]
@natalie-perlin - Looking at changes to `ush/get_crontab_contents.py` and `tests/test_python/test_get_crontab_contents.py` from PR #871, it is possible that by merging the current HEAD of `develop` into the `develop_derecho` branch, the `test_get_crontab_contents` test successfully passes.  However, since PR #889 needs to be merged first, I understand if you would like to wait until that PR has been merged before updating syncing.

File: ush/get_crontab_contents.py
Comment (mkavulich) [inline ush/get_crontab_contents.py]: @natalie-perlin The reason the unit test is failing is that this is not correct python syntax.

```suggestion
    if MACHINE == "CHEYENNE" or MACHINE == "DERECHO":
```

**Review by mkavulich — CHANGES_REQUESTED** [review]
In addition to my other suggested change, we should remove all the `nodesize:` lines from all the files in `parm/wflow/`. It turns out this `<nodesize>` tag in the Rocoto XML actually does nothing without a corresponding `<cores>` tag, which we do not have. And the newer Rocoto build on Derecho gives a bunch of deprecation warnings for this tag each time you run rocotorun, so we should just get rid of it.

Negative news aside, I did confirm I was able to run tests successfully on Derecho! So hopefully once these changes are addressed and the latest development merged in this will be good to go.

File: tests/WE2E/machine_suites/coverage.derecho
Comment (mkavulich) [inline tests/WE2E/machine_suites/coverage.derecho]: Sorry for not catching this earlier...this should be the same list as coverage.cheyenne, no? Since EPIC is not testing on Cheyenne anymore I thought the idea was to move those tests to Derecho.

File: tests/WE2E/machine_suites/comprehensive.derecho
Comment (mkavulich) [inline tests/WE2E/machine_suites/comprehensive.derecho]: This can be the a symbolic link to comprehensive.orion, just like comprehensive.gaea (as of PR #871). Unlike Cheyenne, Derecho's compute nodes have access to the internet. I tested the 3 missing tests from comprehensive.cheyenne (`2020_CAD`, `get_from_AWS_ics_GEFS_lbcs_GEFS_fmt_grib2_2022040400_ensemble_2mems`, and `get_from_NOMADS_ics_FV3GFS_lbcs_FV3GFS`) and they all passed.

File: etc/lmod-setup.csh
Comment (natalie-perlin) [reply etc/lmod-setup.csh]: Changed it to 
if MACHINE == "CHEYENNE" or MACHINE == "DERECHO":

File: tests/WE2E/machine_suites/coverage.derecho
Comment (natalie-perlin) [reply tests/WE2E/machine_suites/coverage.derecho]: @mkavulich - done, thank you for the comment

**Comment by natalie-perlin** [comment]
@mkavulich - addressed your comments on yaml files in wflow/ directory

**Review by mkavulich — APPROVED** [review]
Sorry about those late comments, thanks for addressing them!

**Comment by natalie-perlin** [comment]
Merged changes from develop, and tested without additional cmake options file for UFS WM. After fixing a default for **EXTRN_MDL_DATA_STORES: aws** in ./ush/machine/derecho.yaml, all the fundamental test have passed.

(before correcting derecho.yaml):
```
All 7 experiments finished
Calculating core-hour usage and printing final summary
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              18.17
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  DEAD                   0.00
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE              21.46
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              27.50
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR          COMPLETE              33.77
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0              COMPLETE              31.37
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE              46.74
----------------------------------------------------------------------------------------------------
Total                                                              DEAD                 179.01

Detailed summary written to /glade/derecho/scratch/nperlin/SRW/expt_dirs/WE2E_summary_20230915113527.txt
```

after correcting derecho.yaml:
```
All 1 experiments finished
Calculating core-hour usage and printing final summary
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              22.87
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE              22.87

Detailed summary written to /glade/derecho/scratch/nperlin/SRW/expt_dirs/NCO/WE2E_summary_20230915124841.txt
```

**Comment by natalie-perlin** [comment]
Running comprehensive tests now on Derecho.

**Comment by natalie-perlin** [comment]
@MichaelLueken - are there any additional tests needed for Derecho? As to CI/CD we may not have the account yet.

**Comment by natalie-perlin** [comment]
Comprehesive tests:
```
All 62 experiments finished
Calculating core-hour usage and printing final summary
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
2020_CAD                                                           COMPLETE              33.56
community                                                          COMPLETE              39.81
custom_ESGgrid                                                     COMPLETE              14.23
custom_ESGgrid_Central_Asia_3km                                    DEAD                   0.61
custom_ESGgrid_IndianOcean_6km                                     COMPLETE              21.27
custom_ESGgrid_NewZealand_3km                                      DEAD                   0.71
custom_ESGgrid_Peru_12km                                           COMPLETE              21.60
custom_ESGgrid_SF_1p1km                                            COMPLETE             145.00
custom_GFDLgrid__GFDLgrid_USE_NUM_CELLS_IN_FILENAMES_eq_FALSE      COMPLETE              10.07
custom_GFDLgrid                                                    COMPLETE               9.64
deactivate_tasks                                                   COMPLETE               0.74
get_from_AWS_ics_GEFS_lbcs_GEFS_fmt_grib2_2022040400_ensemble_2me  COMPLETE             636.35
get_from_NOMADS_ics_FV3GFS_lbcs_FV3GFS                             COMPLETE              12.40
grid_CONUS_25km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16      COMPLETE              16.40
grid_CONUS_3km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta   DEAD                   1.96
grid_RRFS_AK_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot        DEAD                   0.66
grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR                 COMPLETE             172.99
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP              COMPLETE              30.66
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot     COMPLETE              34.19
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR             COMPLETE              30.51
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta      COMPLETE              30.70
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE              11.95
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot     COMPLETE              25.66
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              25.64
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR             COMPLETE              36.98
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP              COMPLETE              71.57
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta      COMPLETE              38.72
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP                 COMPLETE              17.23
grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2        COMPLETE              10.89
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE              42.13
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_RRFS_v1beta            COMPLETE              34.03
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2         COMPLETE             211.87
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson  COMPLETE             300.27
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16           COMPLETE             306.42
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR              COMPLETE             320.71
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta       COMPLETE             318.94
grid_RRFS_CONUScompact_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16   COMPLETE              30.74
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR           COMPLETE              25.93
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              25.33
grid_RRFS_CONUScompact_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16   COMPLETE              19.45
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR          COMPLETE              29.55
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              16.47
grid_RRFS_CONUScompact_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16    COMPLETE             251.78
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_HRRR            COMPLETE             264.03
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta     COMPLETE             263.06
grid_RRFS_NA_13km_ics_FV3GFS_lbcs_FV3GFS_suite_RAP                 COMPLETE              75.60
grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0         COMPLETE              30.48
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_HRRR_suite_HRRR                COMPLETE              38.02
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0              COMPLETE              30.28
grid_SUBCONUS_Ind_3km_ics_NAM_lbcs_NAM_suite_GFS_v16               COMPLETE              48.94
grid_SUBCONUS_Ind_3km_ics_RAP_lbcs_RAP_suite_RRFS_v1beta_plot      COMPLETE              15.62
MET_ensemble_verification_only_vx                                  COMPLETE               0.68
MET_verification_only_vx                                           COMPLETE               0.14
nco                                                                COMPLETE              19.69
nco_ensemble                                                       COMPLETE             102.45
nco_grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16      COMPLETE              29.47
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              22.35
nco_grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thom  COMPLETE             292.29
nco_grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR       DEAD                   1.03
pregen_grid_orog_sfc_climo                                         COMPLETE              12.30
specify_EXTRN_MDL_SYSBASEDIR_ICS_LBCS                              COMPLETE              10.88
specify_template_filenames                                         COMPLETE              13.38
----------------------------------------------------------------------------------------------------
```

**Comment by MichaelLueken** [comment]
@natalie-perlin - With the decommissioning of Cheyenne, using the `coverage.cheyenne` test suite for Derecho is perfectly fine.  No additional tests need to be added for Derecho with this PR.

Are there plans to add GNU to Derecho at a later time?  If there are plans, then we can bring in the `coverage.cheyenne.gnu` tests as `coverage.derecho.gnu`.

I'm wrapping up my testing of the Jenkins build and run scripts to ensure that the SRW will build and run using these on Derecho.  Additionally, this will also test the coverage suite for the machine.  Once they pass, I will give my approval and test the rest of the systems using Jenkins.

**Review by MichaelLueken — APPROVED** [review]
@natalie-perlin - The SRW App successfully builds on Derecho using the Jenkins `.cicd/scripts/srw_build.sh` script.  Additionally, the `coverage.derecho` tests were successfully run using `.cicd/scripts/srw_test.sh` and all tests successfully passed:

```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
custom_ESGgrid_IndianOcean_6km                                     COMPLETE              21.29
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot     COMPLETE              35.41
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE              42.15
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR           COMPLETE              26.62
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE              16.56
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_HRRR_suite_HRRR                COMPLETE              38.69
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              23.09
pregen_grid_orog_sfc_climo                                         COMPLETE              12.96
specify_template_filenames                                         COMPLETE              14.32
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             231.09
```
Approving this PR now and running the Jenkins tests for the rest of the platforms (since there is no Jenkins runner for Derecho at this time).

**Comment by MichaelLueken** [comment]
The Jenkins Hera Intel WE2E coverage tests failed for `custom_ESGgrid_Central_Asia_3km`:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
custom_ESGgrid_Central_Asia_3km                                    DEAD                   4.73
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2019061200          COMPLETE               5.59
get_from_HPSS_ics_GDAS_lbcs_GDAS_fmt_netcdf_2022040400_ensemble_2  COMPLETE             751.95
get_from_HPSS_ics_HRRR_lbcs_RAP                                    COMPLETE              13.56
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE               5.77
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot     COMPLETE              12.29
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP                 COMPLETE               9.43
grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2        COMPLETE               6.10
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2         COMPLETE             227.80
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16           COMPLETE             299.96
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_HRRR            COMPLETE             348.04
pregen_grid_orog_sfc_climo                                         COMPLETE               6.15
----------------------------------------------------------------------------------------------------
Total                                                              DEAD                1691.37
```
It failed with a strange NetCDF failure:

`FATAL from PE    18: NetCDF: Start+count exceeds dimension bound: netcdf_read_data_2d: file:INPUT/gfs_data.nc- variable:ps`

A rerun of the test was successful:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
custom_ESGgrid_Central_Asia_3km                                    COMPLETE              23.72
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE              23.72
```
The Orion and Gaea Jenkins tests have successfully passed.  Awaiting completion of Hera GNU and Jet tests now.

**Comment by MichaelLueken** [comment]
Both the Hera GNU and Jet WE2E coverage tests successfully passed on Jenkins.  Now moving forward with merging this work.

**Pull request merged**
